timemachine/fe/geometry.py

```diff
@@ -0,0 +1,67 @@
+# Utility functions to help assign and identify local geometry points
+
+from enum import Enum
+
+from rdkit import Chem
+
+
+class LocalGeometry(Enum):
+    G1_TERMINAL = 0  # R-X
+    G2_KINK = 1  # R-X-H
+    G2_LINEAR = 2  # R-X#N
+    G3_PLANAR = 3  # R-X(=O)O
+    G3_PYRAMIDAL = 4  # R-X(-H)H
+    G4_TETRAHEDRAL = 5  # R-X(-H)(-H)H
+
+
+def assign_atom_geometry(atom):
+    """
+    Heuristic using hybridization information to assign local description
+    of geometry.
+    """
+    nbrs = list(atom.GetNeighbors())
+    if len(nbrs) == 0:
+        assert 0, "Ion not supported"
+    elif len(nbrs) == 1:
+        return LocalGeometry.G1_TERMINAL
+    elif len(nbrs) == 2:
+        if atom.GetHybridization() == Chem.HybridizationType.SP3:
+            return LocalGeometry.G2_KINK
+        elif atom.GetHybridization() == Chem.HybridizationType.SP2:
+            return LocalGeometry.G2_KINK
+        elif atom.GetHybridization() == Chem.HybridizationType.SP:
+            return LocalGeometry.G2_LINEAR
+        else:
+            assert 0, "Unknown 2-nbr geometry!"
+    elif len(nbrs) == 3:
+        if atom.GetHybridization() == Chem.HybridizationType.SP3:
+            return LocalGeometry.G3_PYRAMIDAL
+        elif atom.GetHybridization() == Chem.HybridizationType.SP2:
+            return LocalGeometry.G3_PLANAR
+        else:
+            assert 0, "Unknown 3-nbr geometry"
+    elif len(nbrs) == 4:
+        if atom.GetHybridization() == Chem.HybridizationType.SP3:
+            return LocalGeometry.G4_TETRAHEDRAL
+    else:
+        assert 0, "Too many neighbors"
```

Comment:
Nit: the `len(nbrs) == 4` branch is missing a final `else: assert 0, "unknown 4-nbr geometry"`, will instead print `"Too many neighbors"`

Reply:
great catch. will fix